src/server/v2.0/handler/artifact.go

```diff
@@ -85,13 +85,22 @@ func (a *artifactAPI) ListArtifacts(ctx context.Context, params operation.ListAr
 	if err := a.RequireProjectAccess(ctx, params.ProjectName, rbac.ActionList, rbac.ResourceArtifact); err != nil {
 		return a.SendError(ctx, err)
 	}
+	repositoryName := fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName)
+	_, err := a.repoCtl.GetByName(ctx, repositoryName)
+	if err != nil {
+		switch {
+		case errors.IsErr(err, errors.NotFoundCode):
+			return operation.NewListArtifactsNotFound().WithPayload(&models.Errors{Errors: []*models.Error{{Message: "Repository not found"}}})
```

Comment:
we should use the error library to unify the error handling.

Like, gives a errors.New(nil).WithCode(errors.NotFoundCode).WithMessage("Repository not found")